connectors/embeddings-vector-database/src/main/java/io/camunda/connector/model/embedding/vector/store/AzureCosmosDbNoSqlVectorStore.java

```diff
@@ -0,0 +1,202 @@
+            constraints = @TemplateProperty.PropertyConstraints(notEmpty = true))
+        String endpoint,
+    @Valid @NotNull AzureAuthentication authentication,
+    @NotBlank
+        @TemplateProperty(
+            group = "embeddingsStore",
+            id = "azureCosmosDbNoSql.databaseName",
+            label = "Database name",
+            description = "Specify the name of the Azure Cosmos DB NoSQL database.",
+            constraints = @TemplateProperty.PropertyConstraints(notEmpty = true))
+        String databaseName,
+    @NotBlank
+        @TemplateProperty(
+            group = "embeddingsStore",
+            id = "azureCosmosDbNoSql.containerName",
+            label = "Container name",
+            description = "Specify the name of the Azure Cosmos DB NoSQL container.",
+            constraints = @TemplateProperty.PropertyConstraints(notEmpty = true))
+        String containerName,
+    @NotBlank
+        @TemplateProperty(
+            group = "embeddingsStore",
+            id = "azureCosmosDbNoSql.consistencyLevel",
+            label = "Consistency level",
+            description = "Specify the consistency level for the Azure Cosmos DB NoSQL store.",
+            feel = Property.FeelMode.required,
+            type = TemplateProperty.PropertyType.Dropdown,
+            defaultValue = "EVENTUAL",
+            constraints = @TemplateProperty.PropertyConstraints(notEmpty = true))
+        AzureCosmosDbNoSqlVectorStore.CosmosConsistencyLevel consistencyLevel,
+    @NotBlank
+        @TemplateProperty(
+            group = "embeddingsStore",
+            id = "azureCosmosDbNoSql.distanceFunction",
+            label = "Distance function",
+            description =
+                "The metric used to compute distance/similarity. Details in the <a href=\"https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/vector-search\" target=\"_blank\">documentation</a>.",
+            feel = Property.FeelMode.required,
+            type = TemplateProperty.PropertyType.Dropdown,
+            defaultValue = "COSINE",
+            constraints = @TemplateProperty.PropertyConstraints(notEmpty = true))
+        DistanceFunction distanceFunction,
+    @NotBlank
+        @TemplateProperty(
+            group = "embeddingsStore",
+            id = "azureCosmosDbNoSql.vectorIndexType",
+            label = "Vector index type",
+            description =
+                "The type of vector index type to use. Details in the <a href=\"https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/vector-search\" target=\"_blank\">documentation</a>.",
+            feel = Property.FeelMode.required,
+            type = TemplateProperty.PropertyType.Dropdown,
+            defaultValue = "FLAT",
+            constraints = @TemplateProperty.PropertyConstraints(notEmpty = true))
+        IndexType vectorIndexType)
```

Comment:
💭 Do you think it would make sense to add some tooltips or links to our own documentation outlining this more advanced properties and recommendations what to use?

Reply:
I think referring to the MS documents is a better idea as these are rather technical concepts that are explained better over there.

Reply:
I agree, I meant more providing some additional context in our tooltips or docs to outline what to use in which case. But we can keep it this way and adapt when we know what we can suggest and how often this comes up as question.